### DISCUSSION TITLE
unregister doesnt clear component value

### DISCUSSION THREAD

**Discussion by niralparmar** [react-hook-form/react-hook-form#3884] [Q&A]
Hi,
i have one text field which i am manually unregistering. it is successfully getting unregister and data is excluded from formdata, however user entered value still stays in the text field. i am expceting value also get cleared from component as well. i even tried

> setValue('fieldName',"")

is not working. not sure if i am doing something wrong. 
so if i re register my text field and trigger validation, you will see required field validation but value is still present in text field 

here is my [CSB](https://codesandbox.io/s/react-hook-form-unregister-v6-forked-19446?file=/src/index.js)

i would appreciate any help

**Comment by niralparmar** [comment]
i would appreciate any help , thanks

**Comment by bluebill1049** [comment]
you are register at useEffect, so they r just virtual input, there is no HTML reference holds at hook form, hence the input value isn't getting reset.

**Reply by niralparmar** [reply]
thanks @bluebill1049  appreciate you reply. 
so what it means, i have to provide  

>  ref{register{required:true}}

at the input , and after that i can do register/unregister as per my condition, am i right?

**Reply by bluebill1049** [reply]
`<input ref={register} />` you need to register with the input.

**Reply by niralparmar** [reply]
just updated my [CSB](https://codesandbox.io/s/react-hook-form-unregister-v6-forked-19446?file=/src/index.js) as per your suggestion, but it seems problem still persist.

thanks once again

**Reply by bluebill1049** [reply]
The easiest is to make that component umount, it will sort to the unregister for you. 

try below: 
```
const [show, toggle] = useState();

show && 
```

**Reply by niralparmar** [reply]
thx @bluebill1049 , unfortunately i cant hide input, it just needs to be disabled. is there any other way to achieve this?

**Reply by niralparmar** [reply]
even this official example [CSB](https://codesandbox.io/s/react-hook-form-unregister-v6-9nyzn), it is not resetting the input values. 

is it a bug or expected behavior  that "unregister" will not clear the input.

thanks @bluebill1049 for your help so far.

**Reply by bluebill1049** [reply]
`unregsiter` is made for clear the reference of that input, if you want to clear the input use setValue.
